handlers/mapping_handlers.go

```diff
@@ -0,0 +1,404 @@
+package handlers
+
+import (
+	"encoding/json"
+	"ether/models"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"strconv"
+	"time"
+
+	"github.com/go-sql-driver/mysql"
+	"github.com/gorilla/mux"
+)
+
+func parseMappingJSON(w http.ResponseWriter, body io.ReadCloser, bodyObj *models.UserConversationMapping) error {
+	bodyBytes, err := ioutil.ReadAll(body)
+	if err != nil {
+		errMsg := "Failed to read request body: " + err.Error()
+		log.Println(errMsg)
+		http.Error(w, errMsg, http.StatusBadRequest)
+		return err
+	}
+
+	if err := json.Unmarshal(bodyBytes, bodyObj); err != nil {
+		errMsg := "Failed to parse request body: " + err.Error()
+		log.Println(errMsg)
+		http.Error(w, errMsg, http.StatusBadRequest)
+		return err
+	}
+
+	return nil
+}
+
+// PostMappingHandler adds a single user to a conversation
+func (env *Env) PostMappingHandler(w http.ResponseWriter, r *http.Request) {
+	defer r.Body.Close()
+	userID, err := strconv.ParseInt(r.Header.Get("User-ID"), 10, 64)
+	if err != nil {
+		errMsg := "Invalid user ID"
+		log.Println(errMsg + ": " + err.Error())
+		http.Error(w, errMsg, http.StatusBadRequest)
+		return
+	}
+
+	vars := mux.Vars(r)
+	conversationID, err := strconv.ParseInt(vars["conversation_id"], 10, 64)
+	if err != nil {
+		errMsg := "Invalid conversation ID"
+		log.Println(errMsg + ": " + err.Error())
+		http.Error(w, errMsg, http.StatusBadRequest)
+		return
+	}
+
+	conversation, err := env.getConversation(w, conversationID)
+	if err != nil || conversation == nil {
+		return
+	}
+
+	sessionMember, err := env.DB.GetUserConversationMapping(userID, conversationID)
+	if err != nil {
+		internalServerError(w, err)
+		return
+	}
+	if sessionMember == nil {
+		errMsg := fmt.Sprintf("User %d is not in conversation %d", userID, conversationID)
+		log.Println(errMsg)
+		http.Error(w, "Conversation not found", http.StatusNotFound)
+		return
+	}
+
+	reqMember := &models.UserConversationMapping{}
+	if err := parseMappingJSON(w, r.Body, reqMember); err != nil {
+		return
+	}
+
+	if reqMember.UserID == 0 || reqMember.Role == "" {
+		errMsg := "Request body is missing field(s)"
+		log.Println(errMsg)
+		http.Error(w, errMsg, http.StatusBadRequest)
+		return
+	}
+
+	if reqMember.Role != models.Admin && reqMember.Role != models.User ||
```

Comment:
Might be better to use the `Valid` method for the `reqMember.Role` here and then check if the members roles follow our rules after to make it more clear/readable.